lambda/src/main/scala/pricemigrationengine/handlers/SalesforceAmendmentUpdateHandler.scala

```diff
@@ -1,46 +1,59 @@
 package pricemigrationengine.handlers
 
-import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
 import pricemigrationengine.model.CohortTableFilter.{AmendmentComplete, AmendmentWrittenToSalesforce}
-import pricemigrationengine.model.{CohortItem, Failure, SalesforcePriceRise, SalesforcePriceRiseWriteFailure}
+import pricemigrationengine.model._
 import pricemigrationengine.services._
 import zio.clock.Clock
-import zio.console.Console
-import zio.{ExitCode, IO, Runtime, ZEnv, ZIO, ZLayer}
+import zio.{IO, ZEnv, ZIO, ZLayer}
 
 /**
   * Updates Salesforce with evidence of the price-rise amendment that was applied in Zuora.
   */
-object SalesforceAmendmentUpdateHandler extends zio.App with RequestHandler[Unit, Unit] {
+object SalesforceAmendmentUpdateHandler extends CohortHandler {
 
-  private val main: ZIO[CohortTable with SalesforceClient with Clock with Logging, Failure, Unit] =
+  // TODO: move to config
+  private val batchSize = 1000
+
+  private def main(
+      cohortSpec: CohortSpec
+  ): ZIO[CohortTable with SalesforceClient with Clock with Logging, Failure, HandlerOutput] =
     for {
-      cohortItems <- CohortTable.fetch(AmendmentComplete, None)
-      _ <- cohortItems.foreach(
-        item =>
-          updateSfWithNewSubscriptionId(item).tapBoth(
-            e => Logging.error(s"Failed to update price rise record for ${item.subscriptionName}: $e"),
-            _ => Logging.info(s"Amendment of ${item.subscriptionName} recorded in Salesforce")
-        ))
-    } yield ()
+      cohortItems <- fetchFromCohortTable
+      _ <-
+        cohortItems
+          .take(batchSize)
+          .foreach(item =>
+            updateSfWithNewSubscriptionId(item).tapBoth(
+              e => Logging.error(s"Failed to update price rise record for ${item.subscriptionName}: $e"),
+              _ => Logging.info(s"Amendment of ${item.subscriptionName} recorded in Salesforce")
+            )
+          )
+      itemsToGo <- fetchFromCohortTable
+      numItemsToGo <- itemsToGo.take(1).runCount
+    } yield HandlerOutput(isComplete = numItemsToGo == 0)
```

Comment:
Not sure if this is better, in some rare cases the lambda might get re-executed unnecessarily, but I will save re-querying the table to check for more items.  

```
      count <-
        cohortItems
          .take(batchSize)
          .map(item =>
            updateSfWithNewSubscriptionId(item).tapBoth(
              e => Logging.error(s"Failed to update price rise record for ${item.subscriptionName}: $e"),
              _ => Logging.info(s"Amendment of ${item.subscriptionName} recorded in Salesforce")
            )
          )
          .runCount
    } yield HandlerOutput(isComplete = count < batchSize)
```

Reply:
That is better!  Thanks @frj